src/merge.js

```diff
@@ -0,0 +1,60 @@
+'use strict'
+
+const isPlainObject = require('./isPlainObject')
+const cloneDeep = require('./cloneDeep')
+
+const _cloneObj = (obj) => {
+  try {
+    return cloneDeep(obj)
+  } catch (error) {
+    // If the stringify fails due to circular reference, the merge defaults
+    return Object.assign({}, obj)
+  }
+}
+
+/**
+ * This method is like `assign` except that it recursively merges own and
+ * inherited enumerable string keyed properties of source objects into the
+ * destination object.
+ * @param {Object} target The destination target.
+ * @param {...Object} [sources] The source objects.
+ * @returns {Object} Returns a clone of `target`.
+ */
+function merge (target, ...sources) {
+  if (!isPlainObject(target) || !Array.isArray(sources)) return sources
+
+  const cloneObj = _cloneObj(target)
+
+  for (const i in sources) {
```

Comment:
sorry about this, but wouldn't `for (const source of sources)` be better in this case since we don't care about index?